fem/quadinterpolator.hpp

```diff
@@ -53,38 +54,43 @@ class QuadratureInterpolator
    static const int MAX_ND3D = 1000;
    static const int MAX_VDIM3D = 3;
 
-public:
    enum EvalFlags
    {
       VALUES       = 1 << 0,  ///< Evaluate the values at quadrature points
       DERIVATIVES  = 1 << 1,  ///< Evaluate the derivatives at quadrature points
       /** @brief Assuming the derivative at quadrature points form a matrix,
           this flag can be used to compute and store their determinants. This
           flag can only be used in Mult(). */
-      DETERMINANTS = 1 << 2
+      DETERMINANTS = 1 << 2,
+      PHYSICAL_DERIVATIVES = 1 << 3 ///< Evaluate the physical derivatives
    };
 
    QuadratureInterpolator(const FiniteElementSpace &fes,
-                          const IntegrationRule &ir);
+                          const IntegrationRule &ir,
+                          const bool use_tensor_products = false);
 
    QuadratureInterpolator(const FiniteElementSpace &fes,
-                          const QuadratureSpace &qs);
+                          const QuadratureSpace &qs,
+                          const bool use_tensor_products = false);
 
    /** @brief Disable the use of tensor product evaluations, for tensor-product
        elements, e.g. quads and hexes. */
-   /** Currently, tensor product evaluations are not implemented and this method
-       has no effect. */
-   void DisableTensorProducts(bool disable = true) const
-   { use_tensor_products = !disable; }
+   void DisableTensorProducts() const { use_tensor_products = false; }
+
+   /** @brief Enable the use of tensor product evaluations, for tensor-product
+       elements, e.g. quads and hexes. */
+   void EnableTensorProducts() const { use_tensor_products = true; }
+
+   /** @brief Query the current evaluation mode. */
+   bool UseTensorProducts() const { return use_tensor_products; }
```

Comment:
`UseTensorProducts()` sounds like you are telling the object to use tensor products. Maybe `UsingTensorProducts()`, or `IsUsingTensorProducts()`? Any other suggestions?

Reply:
@v-dobrev : `UsesTensorProducts`, like `fespace` UsesTensorBasis?

Reply:
`UsesTensorProducts` sounds good to me.